src/ng_package/angular_package_format.bzl

```diff
@@ -228,68 +220,33 @@ def _angular_package_format_impl(ctx):
-            # we need to filter out generated files and need to be able to iterate through
-            # typing files in order to determine the entry-point type file.
-            unscoped_types = unscoped_types_depset.to_list()
-
-            # Note: Using `to_list()` is expensive but we cannot get around this here as
-            # we need to filter out generated files to be able to detect entry-point index
-            # files when no flat module metadata is available.
-            unscoped_esm2022_list = unscoped_esm2022_depset.to_list()
-
-            # In case the dependency is built through the "ts_library" rule, or the "ng_module"
-            # rule does not generate a flat module bundle, we determine the index file and
-            # typings entry-point through the most reasonable defaults (i.e. "package/index").
-            es2022_entry_point = _find_matching_file(
-                unscoped_esm2022_list,
-                [
-                    "%s/index.mjs" % entry_point_package,
-                    # Fallback for `ts_project` support where `.mjs` is not auto-generated.
-                    "%s/index.js" % entry_point_package,
-                ],
-            )
-            typings_entry_point = _find_matching_file(unscoped_types, ["%s/index.d.ts" % entry_point_package])
-            guessed_paths = True
-
-            # module_name = es2022_entry_point
-            module_name = es2022_entry_point.short_path[len(owning_package) + 1:][:-(len("index.js") + 1)]
+        # Note: Using `to_list()` is expensive but we cannot get around this here as
+        # we need to filter out generated files and need to be able to iterate through
+        # typing files in order to determine the entry-point type file.
+        unscoped_types = unscoped_types_depset.to_list()
+
+        # Note: Using `to_list()` is expensive but we cannot get around this here as
+        # we need to filter out generated files to be able to detect entry-point index
+        # files when no flat module metadata is available.
+        unscoped_esm2022_list = unscoped_esm2022_depset.to_list()
+
+        # In case the dependency is built through the "ts_library" rule, or the "ng_module"
+        # rule does not generate a flat module bundle, we determine the index file and
+        # typings entry-point through the most reasonable defaults (i.e. "package/index").
+        es2022_entry_point = _find_matching_file(
+            unscoped_esm2022_list,
+            [
+                "%s/index.mjs" % entry_point_package,
+                # Fallback for `ts_project` support where `.mjs` is not auto-generated.
+                "%s/index.js" % entry_point_package,
+            ],
+        )
+        typings_entry_point = _find_matching_file(unscoped_types, ["%s/index.d.ts" % entry_point_package])
+
+        module_name = "/".join([_ for _ in [
+            ctx.attr.package,
+            es2022_entry_point.short_path[len(owning_package) + 1:][:-(len("index.js") + 1)]
```

Comment:
I think the `owning_package` should be sufficient for computing this right? e.g. `dep.label.package` relative to `owning_package` is the entry-point? There is a variable above already (`entry_point`)

Reply:
We need it to be the full npm package name, but the `owning_package` is the bazel package that contains all of this. i.e. `owning_package` would be `packages/angular/ssr`, but `ctx.attr.package` would be `@angular/ssr`

Reply:
Actually was looking at the wrong part of it, we can use that directory reference instead of the package name here.